Cura Definition for Flasforge DreamerNX

KeltE opened the pull request:
Cura Definition for Flasforge DreamerNX

konskarm commented on resources/definitions/flashforge_dreamer_nx.def.json:
```diff
@@ -0,0 +1,48 @@
+        "quality_definition": "flashforge_base",
+        "machine_extruder_trains":
+        {
+            "0": "ff_dreamernx_extruder_0"
```
Minor thing, but please rename the extruder to include the full name of the printer
```suggestion
            "0": "flashforge_dreamernx_extruder_0"
```

Also rename the file `resources/extruders/ff_dreamernx_extruder_0.def.json` to `resources/extruders/flashforge_dreamernx_extruder_0.def.json`

konskarm commented on resources/definitions/flashforge_base.def.json:
```diff
@@ -0,0 +1,44 @@
+        "variants_name": "Nozzle Size"
+    },
+    "overrides": {
+		"machine_heated_bed": {"default_value": true},
```
```suggestion
        "machine_heated_bed": {"default_value": true},
```

konskarm commented on resources/definitions/flashforge_dreamer_nx.def.json:
```diff
@@ -0,0 +1,48 @@
+        "machine_height": {"default_value": 140},
+        "machine_depth": {"default_value": 150},
+        "machine_center_is_zero": {"default_value": true},
+		"machine_heated_bed": {"default_value": true},
```
Mix of tabs and spaces
```suggestion
        "machine_heated_bed": {"default_value": true},
```

konskarm commented on resources/definitions/flashforge_dreamer_nx.def.json:
```diff
@@ -0,0 +1,48 @@
+        "machine_height": {"default_value": 140},
+        "machine_depth": {"default_value": 150},
+        "machine_center_is_zero": {"default_value": true},
+		"machine_heated_bed": {"default_value": true},
+        "machine_max_feedrate_x": { "value": 500 },
+        "machine_max_feedrate_y": { "value": 500 },
+        "machine_max_feedrate_z": { "value": 10 },
+        "machine_max_feedrate_e": { "value": 50 },
+        "machine_max_acceleration_x": { "value": 1500 },
+        "machine_max_acceleration_y": { "value": 1500 },
+        "machine_max_acceleration_z": { "value": 500 },
+        "machine_max_acceleration_e": { "value": 5000 },
+        "machine_acceleration": { "value": 500 },
+        "machine_max_jerk_xy": { "value": 10 },
+        "machine_max_jerk_z": { "value": 0.4 },
+        "machine_max_jerk_e": { "value": 5 },
```
Reduntant values. They are already inherited by the base machine profile.
```suggestion
```

konskarm commented on resources/definitions/flashforge_base.def.json:
```diff
@@ -0,0 +1,44 @@
+        "first_start_actions": ["MachineSettingsAction"],
+        "machine_extruder_trains":
+        {
+            "0": "ff_dreamernx_extruder_0"
```
Minor thing, but please rename the extruder to include the full name of the printer
```suggestion
            "0": "flashforge_dreamernx_extruder_0"
```

Also rename the file `resources/extruders/ff_dreamernx_extruder_0.def.json` to `resources/extruders/flashforge_dreamernx_extruder_0.def.json`

konskarm commented on resources/definitions/flashforge_dreamer_nx.def.json:
```diff
@@ -0,0 +1,48 @@
+        "platform": "FlashForge_DreamerNX.obj",
+        "platform_offset": [0, 0, 0],
+        "quality_definition": "flashforge_base",
+        "machine_extruder_trains":
+        {
+            "0": "ff_dreamernx_extruder_0"
+        }
```
Reduntant, as it is already inherited by the `flashforge_base.def.json`.
```suggestion
```

KeltE replied:
This is my first time. Thank you for pointing out the errors. I'll fix the mistakes tomorrow.
Thanks again!

konskarm commented:
I may have misled you a bit. Along with commit https://github.com/Ultimaker/Cura/pull/9303/commits/d2b55f614cf5467772f1100c01f48930096163ee you also needed to change the actual name of the file `resources/extruders/ff_dreamernx_extruder_0.def.json` into `resources/extruders/flashforge_dreamernx_extruder_0.def.json`. Now the tests cannot find that file.

konskarm reviewed (changes requested):
I may have misled you a bit. Along with commit https://github.com/Ultimaker/Cura/pull/9303/commits/d2b55f614cf5467772f1100c01f48930096163ee you also needed to change the actual name of the file `resources/extruders/ff_dreamernx_extruder_0.def.json` into `resources/extruders/flashforge_dreamernx_extruder_0.def.json`. Now the tests cannot find that file.

Sorry for that, but unfortunately GitHub doesn't allow me to suggest filename changes (AFAIK).

KeltE commented:
I already understood this when I changed your recommendations. I already changed the file name. Hopefully he didn't make any more mistakes. Still confusing Github for me. The terms of all is not yet well understood.

konskarm commented:
Hmm then this change must have not been commited, because the filename is still [`resources/extruders/ff_dreamernx_extruder_0.def.json`](https://github.com/Ultimaker/Cura/pull/9303/files#diff-0ca013d63e37f647a9adcc6cc62ff4e7d2fcafabd4dda72183bbe1dfa8fc0cf2).

KeltE commented:
Maybe I made a mistake.
 I only changed the file name in the printer definition folder. Are changes to my Cura branch not automatically updated?

konskarm commented on resources/definitions/flashforge_dreamer_nx.def.json:
```diff
@@ -0,0 +1,31 @@
+        "file_formats": "text/x-gcode",
+        "platform": "FlashForge_DreamerNX.obj",
+        "platform_offset": [0, 0, 0],
+        "quality_definition": "flashforge_base",
```
I think this comma is the cause of the failing tests.
```suggestion
        "quality_definition": "flashforge_base"
```

KeltE commented:
Thank you very much!  Now the check is successful.

konskarm commented:
Excellent. I also tested it and it slices without issues so I am merging it. The machine will be available to be added in Cura 4.9.

KeltE commented:
Great! Thank you very much for your help.

HadiSDev commented:
@KeltE Willl this one work for the Dreamer as well?

KeltE commented:
> @KeltE Willl this one work for the Dreamer as well?

Unfortunately, there is no experience with a twin-extruder printer, so I can't say exactly. DreamerNX is a single extruder printer. The dreamer then needs to make his own definition file and both extruders his own extruder file.
